src/coreclr/src/jit/gentree.cpp

```diff
@@ -17583,8 +17583,10 @@ CORINFO_CLASS_HANDLE Compiler::gtGetStructHandleIfPresent(GenTree* tree)
                 break;
             }
             case GT_RETURN:
-                structHnd = gtGetStructHandleIfPresent(tree->AsOp()->gtOp1);
+                // We can grab the return type from the 'args' signature in compMethodInfo
+                structHnd = info.compMethodInfo->args.retTypeClass;
```

Comment:
I could back this change out as it is not necessary for this fix

Reply:
Does this change cause the diffs that you see with arm64 multi-reg?

Reply:
No it doesn't cause those diffs.  Occasionally it can give a more specific return type when dealing with generics types.

Reply:
Updated comment:
Yes, it does cause the diffs that I saw. I am backing it out as it is unnecessary